### PR TITLE
[ws-daemon] Git configure safe.directory

### PR DESCRIPTION
**Pull request by aledbf**
## Description

The last git version, v2.34.2 introduced a breaking change, requiring the configuration [safe.directory](https://git-scm.com/docs/git-config/2.35.2#Documentation/git-config.txt-safedirectory). 
Without this setting, `git status` returns an error:
```
git status --porcelain=v2 --branch -uall failed (exit status 128): fatal: unsafe repository ('XXXXXX' is owned by someone else)
To add an exception for this directory, call:

git config --global --add safe.directory XXXXXX
```

## Release Notes
```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/content-service/pkg/git/git.go [hunk 1]
```diff
@@ -169,6 +169,14 @@ func (c *Client) GitWithOutput(ctx context.Context, subcommand string, args ...s
 		env = append(env, fmt.Sprintf("GIT_AUTH_PASSWORD=%s", pwd))
 	}
 
+	homeDir, err := os.MkdirTemp("", "git")
+	if err != nil {
+		return nil, xerrors.Errorf("cannot create temporal directory")
+	}
+	defer os.RemoveAll(homeDir)
+
+	env = append(env, fmt.Sprintf("HOME=%v", homeDir))
```
Comment (geropl) [inline components/content-service/pkg/git/git.go:178]: Hm, I'm not sure I understand. I thought:
 - git would store global config in `~/.gitconfig`
 - if we do not have a shared `HOME` folder across all commands, how is `safe.directory` picked up here? :thinking:

Comment (csweichel) [reply components/content-service/pkg/git/git.go]: it's just needed for the git status. Doesn't matter if this config persists, so long as the commands during the git status collection pick it up.

**Review by csweichel — APPROVED** [review]
Awesome! Thank you.

**Pull request merged**